api/build/compile_publish.go

```diff
@@ -49,7 +49,15 @@ func CompileAndPublish(
 	compiler compiler.Engine,
 	queue queue.Service,
 ) (*pipeline.Build, *models.Item, int, error) {
-	logrus.Debugf("generating queue items for build %s/%d", cfg.Build.GetRepo().GetFullName(), cfg.Build.GetNumber())
+	logger := logrus.WithFields(logrus.Fields{
```

Comment:
same question, why no `c.MustGet` and `l`

Reply:
yea, this is not always gin.Context, so it won't have the logger in it in certain scenarios. opted to just rename to `l` for now. lmk, if you think it's worth doing a type check on context and pulling the logger out if we have it. https://github.com/go-vela/server/pull/1145/commits/924390dfb13b70146a4ddcb0fda4abc9bbe88f16